test/suite/test_truncate14.py

```diff
@@ -0,0 +1,244 @@
+                except WiredTigerError as e:
+                    if wiredtiger_strerror(WT_ROLLBACK) in str(e):
+                        err = WT_ROLLBACK
+                    elif wiredtiger_strerror(WT_PREPARE_CONFLICT) in str(e):
+                        err = WT_PREPARE_CONFLICT
+                    else:
+                        raise e
+                if err != 0:
+                    break
+                scan_cursor.next()
+            scan_cursor.close()
+            del_cursor.close()
+        else:
+            lo_cursor = self.session.open_cursor(uri)
+            hi_cursor = self.session.open_cursor(uri)
+            lo_cursor.set_key(key1)
+            hi_cursor.set_key(key2)
+            try:
+                err = self.session.truncate(None, lo_cursor, hi_cursor, None)
+            except WiredTigerError as e:
+                if wiredtiger_strerror(WT_ROLLBACK) in str(e):
+                    err = WT_ROLLBACK
+                elif wiredtiger_strerror(WT_PREPARE_CONFLICT) in str(e):
+                    err = WT_PREPARE_CONFLICT
+                else:
+                    raise e
+            lo_cursor.close()
+            hi_cursor.close()
+        self.assertEqual(err, 0)
+
+    def check(self, uri, nrows, read_ts, basevalue):
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(read_ts))
+        cursor = self.session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            value = self.mkdata(basevalue, k)
+            self.assertEqual(v, value)
+            count += 1
+        self.session.commit_transaction()
+        self.assertEqual(count, nrows)
+        cursor.close()
+
+    def test_truncate(self):
+        blob_rows = 1000
+        sparse_rows = 20000
+        sparse_gap = 1000000000
+
+        # Create a table without logging.
+        uri = "table:truncate14"
+        ds = SimpleDataSet(
+            self, uri, 0, key_format=self.key_format, value_format=self.value_format,
+            config='log=(enabled=false)' + self.extraconfig)
+        ds.populate()
+
+        if self.value_format == '8t':
+            value_a = 97
+        else:
+            value_a = "aaaaa" * 100
+
+        # Pin oldest and stable timestamps to 10.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(10) +
+            ',stable_timestamp=' + self.timestamp_str(10))
+
+        cursor = self.session.open_cursor(uri)
+
+        self.session.begin_transaction()
+
+        # Write out a dense blob of data at the beginning of the namespace.
+        k = 1
+        n = 0
+        for i in range(1, blob_rows + 1):
+            cursor[k] = self.mkdata(value_a, k)
+            k += 1
+            n += 1
+            # Commit periodically to avoid overflowing the cache.
+            if n % 1009 == 0:
+                self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+                self.session.begin_transaction()
+
```

Comment:
No. Those first three loops are really one stage (note that they have the same commit timestamp) but I think it was an oversight. I think originally they were one big transaction and then I needed to make the test bigger and it didn't fit and I stuck in the periodic commits.

But, I think, maybe it's just as well if the transaction boundaries don't line up with the blob boundaries. It shouldn't make any difference, but if something went off the rails and it did, it'd probably save someone a few minutes of debugging time if things broke at a commit boundary rather than the internal boundaries of the test logic. I'll add a comment. (Added in [9897072](https://github.com/wiredtiger/wiredtiger/pull/8129/commits/9897072051a53d164a8a4989d20f6518126a382a))

Reply:
Ahh fair enough then. Thanks for adding those details.